it/v1/endpoints/AmendPensionsChargesControllerISpec.scala

```diff
@@ -240,36 +205,111 @@ class AmendPensionsChargesControllerISpec extends IntegrationBaseSpec {
+        def errorBody(code: String): String =
+          s"""
+             |{
+             |   "code": "$code",
+             |   "reason": "message"
+             |}
+            """.stripMargin
+
+        val errors = Seq(
+          (BAD_REQUEST, "INVALID_TAXABLE_ENTITY_ID", BAD_REQUEST, NinoFormatError),
+          (BAD_REQUEST, "INVALID_TAX_YEAR", BAD_REQUEST, TaxYearFormatError),
+          (BAD_REQUEST, "INVALID_CORRELATIONID", INTERNAL_SERVER_ERROR, StandardDownstreamError),
+          (BAD_REQUEST, "INVALID_PAYLOAD", BAD_REQUEST, RuleIncorrectOrEmptyBodyError),
+          (UNPROCESSABLE_ENTITY, "REDUCTION_TYPE_NOT_SPECIFIED", INTERNAL_SERVER_ERROR, StandardDownstreamError),
+          (UNPROCESSABLE_ENTITY, "REDUCTION_NOT_SPECIFIED", INTERNAL_SERVER_ERROR, StandardDownstreamError),
+          (INTERNAL_SERVER_ERROR, "SERVER_ERROR", INTERNAL_SERVER_ERROR, StandardDownstreamError),
+          (SERVICE_UNAVAILABLE, "SERVICE_UNAVAILABLE", INTERNAL_SERVER_ERROR, StandardDownstreamError)
         )
-        input.foreach(args => (serviceErrorTest _).tupled(args))
+
+        val extraTysErrors = Seq(
+          (UNPROCESSABLE_ENTITY, "MISSING_ANNUAL_ALLOWANCE_REDUCTION", INTERNAL_SERVER_ERROR, StandardDownstreamError),
+          (UNPROCESSABLE_ENTITY, "MISSING_TYPE_OF_REDUCTION", INTERNAL_SERVER_ERROR, StandardDownstreamError),
+          (UNPROCESSABLE_ENTITY, "TAX_YEAR_NOT_SUPPORTED", BAD_REQUEST, RuleTaxYearNotSupportedError)
+        )
+
+        (errors ++ extraTysErrors).foreach(args => (serviceErrorTest _).tupled(args))
       }
     }
   }
 
+  private trait Test {
+
+    def taxYear: String
+    def downstreamTaxYear: String
+    def downstreamUri: String
+
+    val nino: String = "AA123456A"
+
+    val hateoasResponse: JsValue = Json.parse(
+      s"""
+         |{
+         |   "links":[
+         |      {
+         |         "href":"/individuals/charges/pensions/$nino/$taxYear",
+         |         "method":"GET",
+         |         "rel":"self"
+         |      },
+         |      {
+         |         "href":"/individuals/charges/pensions/$nino/$taxYear",
+         |         "method":"PUT",
+         |         "rel":"create-and-amend-charges-pensions"
+         |      },
+         |      {
+         |         "href":"/individuals/charges/pensions/$nino/$taxYear",
+         |         "method":"DELETE",
+         |         "rel":"delete-charges-pensions"
+         |      }
+         |   ]
+         |}
+          """.stripMargin
+    )
+
+
+    def setupStubs(): StubMapping
+
+    def mtdRequest: WSRequest = {
+      setupStubs()
+      buildRequest(s"/pensions/$nino/$taxYear")
+        .withHttpHeaders(
+          (ACCEPT, "application/vnd.hmrc.1.0+json"),
+          (AUTHORIZATION, "Bearer 123")
+        )
+    }
+
+  }
+
+  private trait NonTysTest extends Test {
+
+    def taxYear: String           = "2021-22"
+    def downstreamTaxYear: String = "2021-22"
```

Comment:
No need for the `downstreamTaxYear` as it is only used in the `downstreamUri` so instead just hard code the year into the uri.